main.js

```diff
@@ -0,0 +1,280 @@
+      'Flip': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(2) > button'),
+      'Percentage': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(3) > button'),
+      'Divide': document.querySelector('.calculator-row:nth-child(1) > .calculator-field:nth-child(4) > button'),
+      'Multiply': document.querySelector('.calculator-row:nth-child(2) > .calculator-field:nth-child(4) > button'),
+      'Minus': document.querySelector('.calculator-row:nth-child(3) > .calculator-field:nth-child(4) > button'),
+      'Plus': document.querySelector('.calculator-row:nth-child(4) > .calculator-field:nth-child(4) > button'),
+      'Equals': document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(3) > button'),
+      'Decimal': document.querySelector('.calculator-row:nth-child(5) > .calculator-field:nth-child(2) > button')
+    },
+    'CurrentInput': document.querySelector('.calculator-current-input'),
+    'History': document.querySelector('.calculator-history')
+  }
+
+  this.updateDisplayNode = function() {
+    var nodeToAdd = this.domElements.CurrentInput.firstChild
+
+    function lineHeightFromFontSize(fontSize, deltaSize) {
+      return ( 2.0 - (deltaSize*1.8) ).toString() // f(x) = 2 - 1.8x
+    }
+
+    if( !nodeToAdd ) {
+      nodeToAdd = document.createElement('p')
+      nodeToAdd.style.margin = 0
+      nodeToAdd.style.display = 'table-cell'
+      this.domElements.CurrentInput.appendChild(nodeToAdd)
+    }
+    nodeToAdd.textContent = this.operands.join('')
+    this.domElements.Operators.Clear.textContent = this.clearOrAllClear === 'all' ? 'AC' : 'C'
+
+    // TODO: Setup dynamic resizing:
+    // Resize line height. Not working at the moment. 
+    // Upon resize, line height is too low, but technically word fits so while loop for reducing
+    // the size, the widths are indeed accurate. (it's vertical overflow thats the problem)
+    // while(this.domElements.CurrentInput.scrollWidth > this.domElements.CurrentInput.parentNode.clientWidth && this.inputSize >= 1.5) {
+    //   this.inputSize -= 0.1
+    //   this.domElements.CurrentInput.style.fontSize = this.inputSize.toString() + 'em'
+    //   this.domElements.CurrentInput.style.lineHeight = lineHeightFromFontSize(this.inputSize, -0.1) + 'em'
+    // }
+    // while(this.domElements.CurrentInput.scrollWidth <= this.domElements.CurrentInput.parentNode.clientWidth && this.inputSize < 3) {
+    //   this.inputSize += 0.1
+    //   this.domElements.CurrentInput.style.fontSize = this.inputSize.toString() + 'em'
+    //   this.domElements.CurrentInput.style.lineHeight = lineHeightFromFontSize(this.inputSize, 0.1) + 'em'
+    // }
+  }
+
+  this.handleNumberInput = function(numberInput) {
+    if( this.domElements.History.children.length > 4 ) {
+      this.domElements.History.removeChild( this.domElements.History.firstChild )
```

Comment:
It's deleting the paragraph that's created from line 77, then building a new one and reinserting it into the dom. Looks like I didn't refactor that out completely after all. I could switch over to just adding it into my HTML then only having to modify it's textContent, if I'm sure it'll be sticking around